crates/astria-conductor/src/executor/client.rs

```diff
@@ -1,154 +1,98 @@
-use astria_core::generated::execution::v1alpha2::{
-    execution_service_client::ExecutionServiceClient,
-    BatchGetBlocksRequest,
-    BatchGetBlocksResponse,
-    Block,
-    BlockIdentifier,
-    CommitmentState,
-    ExecuteBlockRequest,
-    GetBlockRequest,
-    GetCommitmentStateRequest,
-    UpdateCommitmentStateRequest,
+use astria_core::{
+    execution::v1alpha2::{
+        Block,
+        CommitmentState,
+    },
+    generated::execution::{
+        v1alpha2 as raw,
+        v1alpha2::execution_service_client::ExecutionServiceClient,
+    },
+    Protobuf as _,
 };
 use color_eyre::eyre::{
-    Result,
-    WrapErr,
+    self,
+    WrapErr as _,
 };
 use prost_types::Timestamp;
 use tonic::transport::Channel;
 
-use super::ExecutorCommitmentState;
-
-/// Extension trait for the ExecutionServiceClient that makes it easier to interact with the
-/// execution client.
-#[async_trait::async_trait]
-pub(super) trait ExecutionClientExt {
```

Comment:
what's the reasoning for changing this to a wrapper type instead of an extension trait?

Reply:
I started with the extension trait, but you cannot name the trait methods the same as the inherent methods (`execute_block‘ etc... You actually could but it's hecking annoying to use).

So the only way around this is a newtype wrapper.